custom_components/versatile_thermostat/base_thermostat.py

```diff
@@ -147,7 +148,7 @@ def get_tz(hass: HomeAssistant):
     return dt_util.get_time_zone(hass.config.time_zone)
 
 
-class BaseThermostat(ClimateEntity, RestoreEntity):
+class BaseThermostat(ClimateEntity, RestoreEntity, Generic[T]):
```

Comment:
This Generic stuff allows type aides / checkers such as VSCode Pylance or pyright to correctly disambiguate the type of `self._underlyings` as one of `UnderlyingSwitch`, `UnderlyingClimate` or `UnderlyingValve`.

This came about in this PR because VSCode was complaining that `under0.keep_alive_sec` (elsewhere in the PR) was not valid for type `UnderlyingEntity`, which is true as it is only defined for `UnderlyingSwitch`. One could get around it by adding a `keep_alive_sec` property to `UnderlyingEntity`, but this would not be nice semantically as it is really only a property of `UnderlyingSwitch`.